File: 136.cc
#include <functional>
#include <iostream>
#include <numeric>
#include <vector>
using std::accumulate;
using std::cout;
using std::endl;
using std::vector;
using std::bit_xor;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        return accumulate(nums.cbegin(), nums.cend(), 0, bit_xor<int>());
    }
};

int main() {
    vector<int> nums = {1, 1, 2, 3, 3};
    cout << Solution().singleNumber(nums) << endl;
    return 0;
}

File: 136.py
from functools import reduce
import operator

class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return reduce(operator.xor, nums)


if __name__ == "__main__":
    print(Solution().singleNumber([1, 1, 2, 3, 3]))
